javascript/code-challenges/challenges-01.test.js

```diff
@@ -0,0 +1,183 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function named `addOne` that takes an array of numbers, and returns a new array of the numbers, incremented by 1.
+
+Use `forEach` to loop over the input array and work with each value.  Push the new value into a local array. Return the local array;
+------------------------------------------------------------------------------------------------ */
+
+const addOne = (arr) => {
+  const localArray = [];
+  arr.forEach((number) => {
+    localArray.push(number + 1);
+  });
+  return localArray;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named `addExclamation` that takes an array of strings, and returns a new array of the same strings with an "!" added to the end.
+
+Use `forEach` to loop over the input array. Modify each string, and add the updated value into a local array. Return the local array;
+------------------------------------------------------------------------------------------------ */
+
+const addExclamation = (arr) => {
+  // Solution code here...
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named `allUpperCase` that takes an array of strings, and returns a new array of the strings converted to upper case.
+
+Use `forEach` to loop over the input array. The modified strings should each be added into a local array. Return that local array.
+------------------------------------------------------------------------------------------------ */
+
+const allUpperCase = (arr) => {
+  // Solution code here...
+};
+
```

Comment:
A nudge for this one:

We know that forEach executes a callback function once for each element in the arr. Our input here is another arr of strings. Our output is a new array with the strings converted to uppercase.

- We can use forEach to iterate over each string and for each string, (arr.forEach(str =>) we will use the `toUpperCase()` method. We will need to somehow get those new values into a new arr. We know that we can add values to a new array by using `push()`.

I hope this helps!